src/foam/nanos/auth/InvoiceHistoryAuthorizer.js

```diff
@@ -0,0 +1,91 @@
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'InvoiceHistoryAuthorizer',
+  implements: ['foam.nanos.auth.Authorizer'],
+
+  documentation: `AuthorizableAuthorizer which checks if user has access to the invoice history record`,
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.dao.ArraySink',
+    'net.nanopay.invoice.model.Invoice',
+    'foam.dao.history.HistoryRecord',
+    'static foam.mlang.MLang.*',
+    'foam.nanos.auth.AuthorizationException',
+  ],
+
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+        User user = (User) x.get("user");
+        if ( user == null ) {
+          throw new foam.nanos.auth.AuthenticationException("User not found.");
+        } else if ( ! "admin".equalsIgnoreCase(user.getGroup()) && ! "system".equalsIgnoreCase(user.getGroup())) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      javaCode: `
+        DAO invoiceDAO = (DAO) x.get("invoiceDAO");
+        User user = (User) x.get("user");
+
+        ArraySink invoiceArraySink = (ArraySink) invoiceDAO.where(
+          EQ(Invoice.ID, ((HistoryRecord) obj).getObjectId())
+        ).select(new ArraySink());
+        Invoice invoice = (Invoice) invoiceArraySink.getArray().get(0);
+        if ( (invoice.getPayeeId() != user.getId()) && (invoice.getPayerId() != user.getId()) ){
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnUpdate',
+      javaCode: `
+        User user = (User) x.get("user");
+        if ( user == null ) {
+          throw new foam.nanos.auth.AuthenticationException("User not found.");
+        } else if ( ! "admin".equalsIgnoreCase(user.getGroup()) && ! "system".equalsIgnoreCase(user.getGroup())) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'authorizeOnDelete',
+      javaCode: `
+        User user = (User) x.get("user");
+        if ( user == null ) {
+          throw new foam.nanos.auth.AuthenticationException("User not found.");
+        } else if ( ! "admin".equalsIgnoreCase(user.getGroup()) && ! "system".equalsIgnoreCase(user.getGroup())) {
+          throw new AuthorizationException();
+        }
+      `
+    },
+    {
+      name: 'checkGlobalRead',
+      javaCode: `
+        String permission = "invoiceHistoryDAO.read.*";
```

Comment:
```suggestion
        String permission = "invoicehistory.read.*";
```

Conventionally we use the model name in all lowercase as the first part of a permission string, not the DAO name.

Also, there's no need to declare a variable for this since it's only used in one place, you can just inline it. You could also put the string in a constant and reference the constant.